src/lib/journey/_utilities/metadata.utilities.ts

```diff
@@ -1,11 +1,17 @@
-import { CallbackType, ConfirmationCallback, type FRCallback } from '@forgerock/javascript-sdk';
+import {
+  CallbackType,
+  ConfirmationCallback,
+  SelectIdPCallback,
+  type FRCallback,
+} from '@forgerock/javascript-sdk';
 
 import type { CallbackMetadata, WidgetStep } from '$journey/journey.interfaces';
 
 const selfSubmittingCallbacks: string[] = [
```

Comment:
Can we make this `as const` so the type is a union of callback types? @cerebrl is this required to be mutable?

Reply:
Kind of. This will eventually be an exposed setting users of the framework can modify to control the behavior of the app or widget.

Reply:
I guess I never made this very clear, so let me put my thoughts here:

My intention with these data structures and the logic used within the utility files is to keep most of the logic on how to render the form out of the components and avoid callbacks having to be aware of, or directly influence, other callbacks. 

My goal is to build up these metadata objects in a very generic fashion as to ensure high reusability and low specificity. Then, pass them as immutable objects into the components, so each component can make decisions about itself and render accordingly.

To configure the behavior, a dev can just add and remove callbacks from particular lists to alter the form's behavior. This avoids the devs from having to know the specifics or internals of callbacks. The only exception is the idea I commented here: https://github.com/cerebrl/forgerock-web-login-framework/pull/34/files#r1035288585.

The reason that one is different is because it's not just the presence of a callback, but the actual internal state of that callback and its relationship with other callbacks that drives the behavior. So, for that, we need for the dev to pass in a function, rather than a static value.

I hope that makes sense.

Reply:
my thought was really, how can we create an array that is as specific as we can be without breaking the "generic" desire of the code.

string[] feels too unsafe. We can perhaps as const this now with the intention of allowing a developer to `Pick` or `Omit` values from this type? 

Maybe i'm reaching a bit though?

Reply:
Fair point. As long as we have a way for developers to easily manage the types in case they modify them, I'm good.

Reply:
Sorry it's unclear whether you're both requesting a change here - what's your conclusion?

Reply:
It's not a blocker to this PR. I just was suggesting adding the following:

```typescript
const selfSubmittingCallbacks = [
  CallbackType.ConfirmationCallback,
  CallbackType.DeviceProfileCallback,
  CallbackType.PollingWaitCallback,
  CallbackType.SelectIdPCallback,
] as const;

export type SelfSubmittingCallbacks = typeof selfSubmittingCallbacks[number]

const userInputCallbacks = [
  CallbackType.BooleanAttributeInputCallback,
  CallbackType.ChoiceCallback,
  CallbackType.ConfirmationCallback,
  CallbackType.KbaCreateCallback,
  CallbackType.NameCallback,
  CallbackType.NumberAttributeInputCallback,
  CallbackType.PasswordCallback,
  CallbackType.ReCaptchaCallback,
  CallbackType.SelectIdPCallback,
  CallbackType.StringAttributeInputCallback,
  CallbackType.TermsAndConditionsCallback,
  CallbackType.ValidatedCreatePasswordCallback,
  CallbackType.ValidatedCreateUsernameCallback,
] as const;

export type UserInputCallbacks = typeof userInputCallbacks[number]

```

This would be a little more restrictive from a type perspective. we would probably need to `as` cast the FRCallback because it isn't a generic type and getType() returns a string but this a bit more safe and makes sure the values aren't mutable at the type level so instead of "String" we would have a unionion of hte callbacks.